cyclonedx/model/bom.py

```diff
@@ -288,19 +291,19 @@ def components(self) -> "SortedSet[Component]":
     def components(self, components: Iterable[Component]) -> None:
         self._components = SortedSet(components)
 
-    def get_component_by_purl(self, purl: Optional[str]) -> Optional[Component]:
+    def get_component_by_purl(self, purl: Optional["PackageURL"]) -> Optional[Component]:
         """
         Get a Component already in the Bom by its PURL
 
         Args:
              purl:
-                Package URL as a `str` to look and find `Component`
```

Comment:
:question:  so instead of fixing the method you redesigned it to a whole new purpose? 
:x:  this would be a breaking change.

Original comment and typing show intention that it was supposed to work with a string input.
why not jusy add the missing string-cast in the comparrison and leave everything else as is?

Reply:
closed via https://github.com/CycloneDX/cyclonedx-python-lib/pull/310#issuecomment-1272302774